### ISSUE TITLE
Using PS Module "Az.EventHub" V1.5.0 not possible for EventHubs with longer names

### ISSUE THREAD

**Issue by andreasschild** [Azure/azure-powershell#12331]
## Description
I installed the PowerShell module [Az V4.3.0](https://www.powershellgallery.com/packages/Az/4.3.0) and with it I got the new [Az EventHub V1.5.0](https://www.powershellgallery.com/packages/Az.EventHub/1.5.0) installed as well.

Now, EventHubs with names longer than 50 characters are no longer supported. Downgrading to [Az EventHub V1.4.3](https://www.powershellgallery.com/packages/Az.EventHub/1.4.3) solves the problem.

This is either a bug or should at least be marked as a breaking change.

## Steps to reproduce

```powershell
Install-Module -Name "Az.EventHub" -RequiredVersion "1.5.0" -Force
Get-AzEventHub -ResourceGroupName "my-rg" -Namespace "my-ns" -EventHub "my-eventhub-with-at-least-fifty-characters-12345678"

# Error Message: Get-AzEventHub : 'eventHubName' exceeds maximum length of '50'.
```

## Solving the problem

```powershell
Uninstall-Module -Name "Az.EventHub" -RequiredVersion "1.5.0" -Force
Install-Module -Name "Az.EventHub" -RequiredVersion "1.4.3" -Force
```

**Comment by dingmeng-xue** [comment]
@andreasschild , thanks for reporting. Could you share debug message after you set `$DebugPreference="Continue"`?

**Comment by andreasschild** [comment]
Sure:

`PS C:\WINDOWS\system32> Get-AzEventHub -ResourceGroupName "spp-default-dev01" -Namespace "spp-vehiclerealtime-eventhub-ns-dev01" -EventHub "spp-default-eventhub-vehiclerealtime-dataenrichment-dev01"
DEBUG: 2:18:16 PM - GetAzureRmEventHub begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 2:18:16 PM - using account id 'CSTX_a_ANDRSC@corpdir.onmicrosoft.com'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'CSTX_a_ANDRSC@corpdir.onmicrosoft.com', environment: 'AzureCloud', tenant: '9652d7c2-1ccf-4940-8151-4a92bd474ed0'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: '9652d7c2-1ccf-4940-8151-4a92bd474ed0', Endpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob',
ResourceClientUri: 'https://management.core.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/9652d7c2-1ccf-4940-8151-4a92bd474ed0/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: '9652d7c2-1ccf-4940-8151-4a92bd474ed0', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: '1950a258-227b-4e31-a9cf-717495945fc2', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: ADAL PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is
running...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: ADAL PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is
running...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: === Token Acquisition started:
 CacheType: null
 Authentication Target: User
 , Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: === Token Acquisition started:
 Authority: https://login.microsoftonline.com/9652d7c2-1ccf-4940-8151-4a92bd474ed0/
 Resource: https://management.core.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: null
 Authentication Target: User

DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.1950922Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 26 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An expired or near expiry token was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An expired or near expiry token was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3121394Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Refreshing access token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3121394Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Refreshing access token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Storing token in the cache...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: Storing token in the cache...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An item was stored in the cache
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: An item was stored in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 06/07/2020 13:18:15 +00:00
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: e434db5f-4fe7-4ee1-bdbc-2f61065064bc - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 06/07/2020 13:18:15 +00:00Access Token Hash:
BhCRvTvw33dHGZR26XJmeuL3YZMk+PtYWp4dv7ICN68=
  User id: 76972289-8044-4dcc-a18c-ae1b2049c9c2
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Serializing token cache with 26 items.
Get-AzEventHub : 'eventHubName' exceeds maximum length of '50'.
At line:1 char:1
+ Get-AzEventHub -ResourceGroupName "spp-default-dev01" -Namespace "spp ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzEventHub], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.EventHub.Commands.EventHub.GetAzureRmEventHub

DEBUG: AzureQoSEvent: CommandName - Get-AzEventHub; IsSuccess - False; Duration - 00:00:00.7686982;; Exception - Microsoft.Rest.ValidationException: 'eventHubName' exceeds maximum length of '50'.
   at Microsoft.Azure.Management.EventHub.EventHubsOperations.<GetWithHttpMessagesAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.EventHub.EventHubsOperationsExtensions.<GetAsync>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.EventHub.EventHubsOperationsExtensions.Get(IEventHubsOperations operations, String resourceGroupName, String namespaceName, String eventHubName)
   at Microsoft.Azure.Commands.Eventhub.EventHubsClient.GetEventHub(String resourceGroupName, String namespaceName, String eventHubName)
   at Microsoft.Azure.Commands.EventHub.Commands.EventHub.GetAzureRmEventHub.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 2:18:17 PM - GetAzureRmEventHub end processing.`

A colleague has the same issue.

**Comment by andreasschild** [comment]
Sure:

`PS C:\WINDOWS\system32> Get-AzEventHub -ResourceGroupName "xxx" -Namespace "xxx" -EventHub "xxx"
DEBUG: 2:18:16 PM - GetAzureRmEventHub begin processing with ParameterSet '__AllParameterSets'.
DEBUG: 2:18:16 PM - using account id 'xxx'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'xxx', environment: 'AzureCloud', tenant: 'xxx'
DEBUG: [Common.Authentication]: Authenticating using configuration values: Domain: 'xxx', Endpoint: 'https://login.microsoftonline.com/', ClientId: 'xxx', ClientRedirect: 'urn:ietf:wg:oauth:2.0:oob',
ResourceClientUri: 'https://management.core.windows.net/', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using context with Authority 'https://login.microsoftonline.com/xxxx/', CorrelationId: '00000000-0000-0000-0000-000000000000', ValidateAuthority: 'True'
DEBUG: [Common.Authentication]: Acquiring token using AdalConfiguration with Domain: 'xxxx', AdEndpoint: 'https://login.microsoftonline.com/', ClientId: 'xxxx', ClientRedirectUri: urn:ietf:wg:oauth:2.0:oob
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: xxx - LoggerBase.cs: ADAL PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is
running...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: xxxx - LoggerBase.cs: ADAL PCL.Desktop with assembly version '3.19.2.6005', file version '3.19.50302.0130' and informational version '2a8bec6c4c76d0c1ef819b55bdc3cda2d2605056' is
running...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: xxx - LoggerBase.cs: === Token Acquisition started:
 CacheType: null
 Authentication Target: User
 , Authority Host: login.microsoftonline.com
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.1950922Z: xxx - LoggerBase.cs: === Token Acquisition started:
 Authority: https://login.microsoftonline.com/xxx/
 Resource: https://management.core.windows.net/
 ClientId: xxx
 CacheType: null
 Authentication Target: User

DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.1950922Z: xxx  - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.1950922Z: xxx  - LoggerBase.cs: Loading from cache.
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Deserialized 26 items to token cache.
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3091158Z:xxx  - LoggerBase.cs: Looking up cache for a token...
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: An item matching the requested resource was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: An expired or near expiry token was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: An expired or near expiry token was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.3091158Z: xxx  - LoggerBase.cs: A matching item (access token or refresh token or both) was found in the cache
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3121394Z: xxx  - LoggerBase.cs: Refreshing access token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.3121394Z: xxx  - LoggerBase.cs: Refreshing access token...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: xxx  - LoggerBase.cs: Storing token in the cache...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: xxx  - LoggerBase.cs: Storing token in the cache...
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: xxx  - LoggerBase.cs: An item was stored in the cache
DEBUG: [ADAL]: Verbose: 2020-07-06T12:18:16.8245155Z: xxx  - LoggerBase.cs: An item was stored in the cache
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: xxx - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 06/07/2020 13:18:15 +00:00
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: xxx - LoggerBase.cs: === Token Acquisition finished successfully. An access token was returned: Expiration Time: 06/07/2020 13:18:15 +00:00Access Token Hash:
xxx8=
  User id: xxx
DEBUG: [ADAL]: Information: 2020-07-06T12:18:16.8265180Z: 00000000-0000-0000-0000-000000000000 - LoggerBase.cs: Serializing token cache with 26 items.
Get-AzEventHub : 'eventHubName' exceeds maximum length of '50'.
At line:1 char:1
+ Get-AzEventHub -ResourceGroupName "xxx" -Namespace "xxx...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzEventHub], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.EventHub.Commands.EventHub.GetAzureRmEventHub

DEBUG: AzureQoSEvent: CommandName - Get-AzEventHub; IsSuccess - False; Duration - 00:00:00.7686982;; Exception - Microsoft.Rest.ValidationException: 'eventHubName' exceeds maximum length of '50'.
   at Microsoft.Azure.Management.EventHub.EventHubsOperations.<GetWithHttpMessagesAsync>d__14.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.EventHub.EventHubsOperationsExtensions.<GetAsync>d__19.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.EventHub.EventHubsOperationsExtensions.Get(IEventHubsOperations operations, String resourceGroupName, String namespaceName, String eventHubName)
   at Microsoft.Azure.Commands.Eventhub.EventHubsClient.GetEventHub(String resourceGroupName, String namespaceName, String eventHubName)
   at Microsoft.Azure.Commands.EventHub.Commands.EventHub.GetAzureRmEventHub.ExecuteCmdlet()
   at Microsoft.WindowsAzure.Commands.Utilities.Common.AzurePSCmdlet.ProcessRecord();
DEBUG: Finish sending metric.
DEBUG: 2:18:17 PM - GetAzureRmEventHub end processing.`

A colleague has the same issue.

**Comment by Bindsi** [comment]
Hey guys,
having the same issue here.
Now the new module Az.EventHub is also rolled out on Azure DevOps agents and blocks us obtaining the PrimaryConString to configure our microservices accessing the EH.

Any fix in progress or update available on this?
Thx

**Comment by andreasschild** [comment]
I just found the resource naming restriction and it says, that there is a limitation of 50 characters:
https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsofteventhub

In portal, there is a error message:
![image](https://user-images.githubusercontent.com/10256547/87436022-c669d600-c5ec-11ea-9e52-f7af0a0553e7.png)

Creating a new Event Hub by PowerShell Az Module also throw an error:
![image](https://user-images.githubusercontent.com/10256547/87437057-1d23df80-c5ee-11ea-95dc-d338d1b25f39.png)

But, we are creating the Event Hub via ARM and there the limitation is 256 characters:
https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas

Long story short: In the docus, there are two limitations of the event hub name defined. Is one of the documentation wrong?

BR

**Comment by samuelkoppes** [comment]
We are investigating this issue, and will report back with an update by 11/4/2020.

**Comment by SaurabhSharma-MSFT** [comment]
@samuelkoppes Do you have any updates on this issue.
@andreasschild Apologies for the delayed response. I see that this issue is opened long time ago and no further activity had taken place. So wanted to check if you are still looking for assistance on this query? Please let us know .

**Comment by andreasschild** [comment]
@SaurabhSharma-MSFT: Thanks for comming back to me. I shorten the name of the event hub to 50 characters. With that, the issue was solved. But it would be good if this is documented anywhere.

**Comment by Saglodha** [comment]
Hi @andreasschild , Apologies for delay in response here. 

I had been reviewing the issue and could conduct some tests, I was able to set the event hub name to be within limit of 256 characters (using Azure Portal, ARM template and PS as well). Below is the snip: 

ARM Template:
![EventhubnaminginARM](https://user-images.githubusercontent.com/62641016/151929131-18678373-a942-4519-b34f-943e86a3320d.png)

Portal:

<img width="473" alt="image" src="https://user-images.githubusercontent.com/62641016/151929457-eed71d41-de27-4c65-819b-edf6a5f6a076.png">

PS:
<img width="830" alt="image" src="https://user-images.githubusercontent.com/62641016/151929515-6fbf4250-243e-46b9-8a53-7a6fdeb899c8.png">

Portal indeed prompts when we surpass the naming limit during validation phase, here is the snip:

<img width="491" alt="image" src="https://user-images.githubusercontent.com/62641016/151929897-341d955c-1ea0-4d45-93ff-d05f62b188de.png">

Thus, the limits enforced on all client experiences look to be in the same (1-256 characters long), I would work with the team internally to get the documentation updated. 


Please feel free to respond if you observe any discrepant results at your end.

**Comment by navba-MSFT** [comment]
@andreasschild When I used the most recent version of the Az.EventHub I didn't face this issue. [This](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas#common-limits-for-all-tiers) article talks about name limit being 256 characters. Could you please test the behavior with the most [recent version](https://www.powershellgallery.com/packages/Az.EventHub/1.9.1) of the Az.EventHub? Awaiting your reply.

**Comment by navba-MSFT** [comment]
@andreasschild [This](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsofteventhub) docs has been updated with the right character length for the event hub name. see [#87316](https://github.com/MicrosoftDocs/azure-docs/pull/87316) Please let us know if you need any further assistance on this issue.

**Comment by andreasschild** [comment]
@navba-MSFT: I tested the method 'Get-AzEventHub' with Az.EventHub Version 1.8.0 and the error message about the limitation of 50 characters are no showing up. With this, I am going to close this issue
